packages/gestalt/src/Button.js

```diff
@@ -142,7 +157,89 @@ const ButtonWithForwardRef: React$AbstractComponent<
     </Text>
   );
 
+  const iconEndComponent = (): Node => {
+    const SIZE_NAME_TO_PIXEL = {
+      sm: 10,
+      md: 12,
+      lg: 12,
+    };
+
+    return (
+      <Box alignItems="center" display="flex">
+        {buttonText}
+        <Box display="inlineBlock" flex="none" marginStart={2}>
+          <Icon
+            accessibilityLabel=""
+            color={textColor}
+            icon={iconEnd}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
   /* eslint-disable react/button-has-type */
+  if (props.type === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        hoverStyle="none"
+        href={href}
+        onClick={({ event }) => {
```

Comment:
@AlbertCarreras could we extract this `onClick` + the one below into 1 `handleClick`?

```jsx
function handleClick(event) {
  if (onClick) {
    onClick({event})
  }
}
///
<Link onClick={handleClick} />
```